txinput.go

```diff
@@ -2,14 +2,31 @@ package bt
 
 import (
 	"bytes"
+	"context"
 	"encoding/binary"
+	"errors"
 	"fmt"
 
 	"github.com/libsv/go-bk/crypto"
 
 	"github.com/libsv/go-bt/v2/bscript"
 )
 
+// FundIteratorFunc is used for AutoFund. It expect bt.FundIteration to be returned
+// containing relevant input information, and a bool informing if the retrieval was successful.
+//
+// It is expected that the boolean return value should return false once funds are depleted.
+type FundIteratorFunc func() (*FundIteration, bool)
```

Comment:
As this will most likely be looking up funds from a data store, add ctx as an argument.

Reply:
Would FundGetterFunc be a better name for this do you think as it explains more what it's doing, getting a single fund.

Reply:
I thought since we're requesting n funds from a collection without any identifying args, leaving state management up to the implementor, iterator was better to describe it.

What do you think?

Reply:
Aye the more I read it as FundGetter the more I agree actually

Reply:
Yeah as each call will get a fund / input